source/CScriptEngine.cpp

```diff
@@ -1,1038 +1,1073 @@
+            CustomMission->SetActive(false);
+            CustomMission = nullptr;
+            *MissionLoaded = false;
+        }
+        else
+        {
+            if (cs->bSaveEnabled)
+            {
+                InactiveScriptHashes.insert(cs->dwChecksum);
+                TRACE("Stopping custom script named %s", cs->Name);
+            }
+            else
+            {
+                TRACE("Unregistering custom script named %s", cs->Name);
+                ScriptsWaitingForDelete.push_back(cs);
+            }
+
+            //TRACE("Psyke!");
+
+            CustomScripts.remove(cs);
+            RemoveScriptFromQueue(cs, activeThreadQueue);
+            cs->SetActive(false);
+
+            /*if(!pScript->IsMission()) *MissionLoaded = false;
+            RemoveScriptFromQueue(pScript, activeThreadQueue);
+            AddScriptToQueue(pScript, inactiveThreadQueue);
+            StopScript(pScript);*/
+        }
+    }
+
+    void CScriptEngine::RemoveAllCustomScripts(void)
+    {
+        InactiveScriptHashes.clear();
+        std::for_each(CustomScripts.begin(), CustomScripts.end(), [this](CCustomScript *cs) {
+            TRACE("Unregistering custom script named %s", cs->Name);
+            RemoveScriptFromQueue(cs, activeThreadQueue);
+            //AddScriptToQueue(cs, inactiveThreadQueue);
+            //if(cs->GetPrev()) cs->GetPrev()->SetNext(nullptr);
+            //if(cs->GetNext()) cs->GetNext()->SetPrev(nullptr);
+            //TRACE("Psyke!!");
+            cs->SetActive(false);
+            delete cs;
+        });
+        CustomScripts.clear();
+        std::for_each(ScriptsWaitingForDelete.begin(), ScriptsWaitingForDelete.end(), [this](CCustomScript *cs) {
+            TRACE("Deleting inactive script named %s", cs->Name);
+            delete cs;
+        });
+        ScriptsWaitingForDelete.clear();
+        if (CustomMission)
+        {
+            TRACE("Unregistering custom mission named %s", CustomMission->Name);
+            RemoveScriptFromQueue(CustomMission, activeThreadQueue);
+            CustomMission->SetActive(false);
+            delete CustomMission;
+            CustomMission = nullptr;
+            *MissionLoaded = false;
+        }
+    }
+
+    void CScriptEngine::UnregisterAllScripts()
+    {
+        TRACE("Unregistering all custom scripts");
+        std::for_each(CustomScripts.begin(), CustomScripts.end(), [this](CCustomScript *cs) {
+            RemoveScriptFromQueue(cs, activeThreadQueue);
+            cs->SetActive(false);
+        });
+    }
+
+    void CScriptEngine::ReregisterAllScripts()
+    {
+        TRACE("Reregistering all custom scripts");
+        std::for_each(CustomScripts.begin(), CustomScripts.end(), [this](CCustomScript *cs) {
+            AddScriptToQueue(cs, activeThreadQueue);
+            cs->SetActive(true);
+        });
+    }
+
+    CCustomScript::CCustomScript(const char *szFileName, bool bIsMiss, CCustomScript *parent, int label)

```

Comment:
I think this should be split on two sub classes:
```
CCustomChildScript(CCustomScript *parent, int label): CCustomScript
CCustomExternalScript(const char *szFileName, bool bIsMiss): CCustomScript
```
so you can't mess with parameters. 

@JuniorDjjr what do you think?

Reply:
I agree, I have to say it was laziness.

Reply:
...On the other hand, it would probably create code duplication.

Reply:
Between code duplication and safety I prefer safety :)